File: internal/templates/templates/api-and-worker/_node/services/create-order/handler.mjs
// POST /orders — save a new order and queue it for processing.
import { randomUUID } from "node:crypto";
import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import { DynamoDBDocumentClient, PutCommand } from "@aws-sdk/lib-dynamodb";
import { SQSClient, GetQueueUrlCommand, SendMessageCommand } from "@aws-sdk/client-sqs";

const db = DynamoDBDocumentClient.from(new DynamoDBClient({}));
const sqs = new SQSClient({});
const { QueueUrl } = await sqs.send(new GetQueueUrlCommand({ QueueName: process.env.QUEUE_NAME }));

export const handler = async (event) => {
  const order = JSON.parse(event.body ?? "{}");
  if (!order.sku) {
    return { statusCode: 422, body: JSON.stringify({ error: "sku is required" }) };
  }

  order.id = randomUUID();
  order.status = "pending";
  order.createdAt = new Date().toISOString();

  await db.send(new PutCommand({ TableName: process.env.TABLE_NAME, Item: order }));
  await sqs.send(new SendMessageCommand({
    QueueUrl,
    MessageBody: JSON.stringify({ id: order.id, fail: order.fail ?? false }),
  }));

  console.log(`order ${order.id} saved and queued`);
  return { statusCode: 201, body: JSON.stringify(order) };
};

File: internal/templates/templates/api-and-worker/_python/services/create-order/handler.py
"""POST /orders — save a new order and queue it for processing."""

import json
import os
import uuid
from datetime import datetime, timezone

import boto3

table = boto3.resource("dynamodb").Table(os.environ["TABLE_NAME"])
sqs = boto3.client("sqs")
queue_url = sqs.get_queue_url(QueueName=os.environ["QUEUE_NAME"])["QueueUrl"]


def handler(event, context):
    order = json.loads(event.get("body") or "{}")
    if not order.get("sku"):
        return {"statusCode": 422, "body": json.dumps({"error": "sku is required"})}

    order["id"] = str(uuid.uuid4())
    order["status"] = "pending"
    order["createdAt"] = datetime.now(timezone.utc).isoformat()

    table.put_item(Item=order)
    sqs.send_message(
        QueueUrl=queue_url,
        MessageBody=json.dumps({"id": order["id"], "fail": order.get("fail", False)}),
    )

    print(f"order {order['id']} saved and queued")
    return {"statusCode": 201, "body": json.dumps(order)}
